elegantthemes/d5-extension-example-modules
# D5 :: Docs :: Add example of Spacing default value as part of decoration property of module attribute element

# The Issue

## Issue Details

### Issue Reference
Fixes: https://github.com/elegantthemes/Divi/issues/41000

### Technical Description

I just realized we don't have real example used for decoration in our module example. So, this is the best chance to add it on our module examples.

## Root Cause Explanation

### Why it Happened

N/A

### Git Blame Insights

N/A

### Prior Code Author Contact

N/A

---

# The Pull Request

## Resolution Details

### PR Solution Explanation

This PR add example of default value for spacing. We have some examples of default value used in https://github.com/elegantthemes/d5-extension-example-modules. However, we need something more related to decoration. Hence this example added.

First, we need to set the spacing default value in `module.json`. Since the spacing that we will target here is part of `decoration` of `module` attribute element, the code will look like this:

```
{
  "name": "example/static-module",
  // Other code ...
  "attributes": {
    "module": {
      "type": "object",
      "selector": "{{selector}}",
      "default": {
        "meta": {
          // Other code...
        },
        "advanced": {
          // Other code...
        },
        "decoration": {
          "spacing": {
            "desktop": {
              "value": {
                "margin": {
                  "top":    "0px",
                  "right":  "auto",
                  "bottom": "30px",
                  "left":   "auto"
                }
              }
            }
          }
        }
      }
    }
  }
}
```

In order to make the default value is applied properly in the Spacing Group settings, let's add the `defaultGroupAttr `to the `SpacingGroup` of `SettingsDesign`:

```
export const SettingsDesign = ({
   defaultSettingsAttrs,
 }: Module.Settings.Panel.Props<StaticModuleAttrs>): ReactElement => (
  <React.Fragment>
    {/* Other code ... */}
    <SpacingGroup
      defaultGroupAttr={defaultSettingsAttrs?.module?.decoration?.spacing?.asMutable({ deep: true }) ?? {}}
    />
  </React.Fragment>
);
```

### Screencast Verification

https://github.com/user-attachments/assets/e3a94fe9-acd4-4afe-bdcd-8a0e0386a8a8

### QA Strategy for Regression Testing

- Install and activate this plugin: [d5-extension-example-modules-v1.0.0.zip](https://github.com/user-attachments/files/18244049/d5-extension-example-modules-v1.0.0.zip). This plugin contains some example modules, one of them is Static Module.
- Open VB and add Static Module.
- Go to Design > Spacing > Margin option.
- Make sure the Margin option has default value, not just empty input.

https://github.com/user-attachments/assets/e3a94fe9-acd4-4afe-bdcd-8a0e0386a8a8

## Additional Considerations (Optional)

### Alternate Solution

N/A

## Miscellaneous Details

### Changelog Copy

Added example of default Spacing default value in Static module as part of decoration property of module attribute element.
